homeassistant/components/enigma2/config_flow.py

```diff
@@ -0,0 +1,158 @@
+        """Validate user input."""
+
+        self.errors = {}
+
+        self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+
+        base_url = URL.build(
+            scheme="http" if not user_input[CONF_SSL] else "https",
+            host=user_input[CONF_HOST],
+            port=user_input[CONF_PORT],
+            user=user_input.get(CONF_USERNAME),
+            password=user_input.get(CONF_PASSWORD),
+        )
+
+        session = async_create_clientsession(
+            self.hass, verify_ssl=user_input[CONF_VERIFY_SSL], base_url=base_url
+        )
+
+        try:
+            about = await OpenWebIfDevice(session).get_about()
+        except InvalidAuthError:
+            self.errors["base"] = "invalid_auth"
+        except ClientError:
+            self.errors["base"] = "cannot_connect"
+        except Exception:  # pylint: disable=broad-except
+            self.errors["base"] = "unknown"
+        else:
+            await self.async_set_unique_id(about["info"]["ifaces"][0]["mac"])
+            self._abort_if_unique_id_configured()
+
+        return user_input
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the user step."""
+        if user_input is None:
+            return self.async_show_form(step_id=SOURCE_USER, data_schema=CONFIG_SCHEMA)
+
+        data = await self.validate_user_input(user_input)
+        if "base" in self.errors:
+            return self.async_show_form(
+                step_id=SOURCE_USER, data_schema=CONFIG_SCHEMA, errors=self.errors
+            )
+        return self.async_create_entry(
+            data=data, title=data[CONF_HOST], options=self._options
+        )
+
+    async def async_step_import(self, user_input: dict[str, Any]) -> ConfigFlowResult:
+        """Validate import."""
+        if CONF_PORT not in user_input:
+            user_input[CONF_PORT] = DEFAULT_PORT
+        if CONF_SSL not in user_input:
+            user_input[CONF_SSL] = DEFAULT_SSL
+        user_input[CONF_VERIFY_SSL] = DEFAULT_VERIFY_SSL
+
+        async_create_issue(
+            self.hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.11.0",
+            is_fixable=False,
+            is_persistent=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "Enigma2",
+            },
+        )
+
+        self._data = {
+            key: user_input[key] for key in user_input if key in self.DATA_KEYS
+        }
+        self._options = {
+            key: user_input[key] for key in user_input if key in self.OPTIONS_KEYS
+        }
+
+        return await self.async_step_user(self._data)
```

Comment:
If this function returns errors, the import flow doesn't like that (don't know exactly what it did, but one of the integrations last month had that), so we should abort if an error happens, and because of ADR-0021, we should let the user know why the import failed. (So when the host could not be found, that should be reflected in the issue message). Currently you always create 1 issue, which should only be raised whenever it imports for the first time, or whenever the entry already exists, otherwise the issue should be a custom one for the cause.